libraries/botbuilder/src/teamsActivityHandler.ts

```diff
@@ -924,4 +928,104 @@ export class TeamsActivityHandler extends ActivityHandler {
+     *
+     * @param context The context for this turn.
+     * @returns A promise that represents the work queued.
+     */
+    protected async onTeamsMeetingStart(context: TurnContext): Promise<void> {
+        await this.handle(context, 'TeamsMeetingStart', this.defaultNextEvent(context));
+    }
+
+    /**
+     * Invoked when a Meeting End event activity is received from the connector.
+     * Override this in a derived class to provide logic for when a meeting is ended.
+     *
+     * @param context The context for this turn.
+     * @returns A promise that represents the work queued.
+     */
+    protected async onTeamsMeetingEnd(context: TurnContext): Promise<void> {
+        await this.handle(context, 'TeamsMeetingEnd', this.defaultNextEvent(context));
+    }
+
+    /**
+     * Registers a handler for when a Teams meeting starts.
+     *
+     * @param handler A callback that handles Meeting Start events.
+     * @returns A promise that represents the work queued.
+     */
+    public onTeamsMeetingStartEvent(
+        handler: (meeting: MeetingStartEventDetails, context: TurnContext, next: () => Promise<void>) => Promise<void>
+    ): this {
+        return this.on('TeamsMeetingStart', async (context, next) => {
+            const meeting = this.convertMeetingEventDetailsToCamelCase<MeetingStartEventDetails>(
+                context.activity.value
+            );
+            await handler(meeting, context, next);
+        });
+    }
+
+    /**
+     * Registers a handler for when a Teams meeting ends.
+     *
+     * @param handler A callback that handles Meeting End events.
+     * @returns A promise that represents the work queued.
+     */
+    public onTeamsMeetingEndEvent(
+        handler: (meeting: MeetingEndEventDetails, context: TurnContext, next: () => Promise<void>) => Promise<void>
+    ): this {
+        return this.on('TeamsMeetingEnd', async (context, next) => {
+            const meeting = this.convertMeetingEventDetailsToCamelCase<MeetingEndEventDetails>(context.activity.value);
+            await handler(meeting, context, next);
+        });
+    }
+
+    // The payload from Teams comes in with TitleCase keys, so we'll convert them to camelCase
+    // to maintain JSON and JS standard practices.
+    private convertMeetingEventDetailsToCamelCase<T extends MeetingStartEventDetails | MeetingEndEventDetails>(
+        meeting: Record<string, string>
+    ): T {
+        const convertedMeeting: MeetingEventDetails = {
+            id: meeting.Id,
+            joinUrl: meeting.JoinUrl,
+            meetingType: meeting.MeetingType,
+            title: meeting.Title,
+        };
+
+        if (meeting.StartTime) {
+            (convertedMeeting as MeetingStartEventDetails).startTime = new Date(meeting.StartTime);
+        } else {
+            (convertedMeeting as MeetingEndEventDetails).endTime = new Date(meeting.EndTime);
+        }
+
+        return convertedMeeting as T;
+    }
```

Comment:
Note: @joshgummersall I ended up going back to this route because it seemed the cleanest. We know the shape of the `meeting` argument, but I'd need to create separate interfaces/types for them that are identical to the existing ones, except for capitalization. I still don't _love_ this and am open to other ideas, but I wanted to avoid introducing a bunch of extra interfaces and this keeps the `convertedMeeting` declaration a little more DRY.